api/thrift/api.thrift

```diff
@@ -259,6 +259,10 @@ struct MetaData {
     14: optional bool historicalBackfill
     // Optional expected deprecation date
     15: optional string deprecationDate
+    // A list of applicable modes related to this config
+    16: optional list<string> applicableModes
+    // A list of related tables that this config generates
+    17: optional list<string> relatedTables
```

Comment:
I wonder if a map<string, string> would be better here, where key is mode and value is table name. This will make it more clear which table belong to which mode, and also removes the need to keep `applicableModes`  

Also, in terms of naming, how about `outputTables`?

Reply:
yes, I think `outputTables` is a better name. I can change this. 

I think the issue with `map<string, string>` (what I went with first), is that for some modes there seem to be multiple output tables.  

For e.g. [label-join](https://sourcegraph.a.musta.ch/github.com/airbnb/chronon/-/blob/api/py/ai/chronon/utils.py?L437-440). I didn't quite understand what the different table names mean here, but I assumed that mode:table was a 1:many mapping.

Reply:
Yeah, another way is to use `map<string, list<string>>`. A more common example of having multiple output tables per job is the `join`, which can generate bootstrap_table, various join_part_table, and final_table. I noticed that `bootstrap_table` and `join_part_table`s are not in your example?

Reply:
@hzding621  Done. Changed to map<string, list<string>>, and merged `bootstrap` into `backfill`. 
Can you help me understand the join_parts tables?